### PR TITLE
Handling of all day events

### PR DESCRIPTION
**Pull request by derekantrican**
Adding handling of all day events

### CODE REVIEW

File: Code.gs [hunk 1]
```diff
@@ -96,6 +96,7 @@ function main(){
   if(targetCalendar == null){
       Logger.log("Creating Calendar: " + targetCalendarName);
       targetCalendar = CalendarApp.createCalendar(targetCalendarName);
+      targetCalendar.setSelected(true); //Sets the calendar as "shown" in the Google Calendar UI
```
Comment (michaelcadilhac) [inline Code.gs]: Isn't that imposing on the user?

File: Code.gs [hunk 2]
```diff
@@ -145,13 +146,25 @@ function main(){
...
-        var resultEvent = targetCalendar.createEvent(event.title, 
-                                                     event.startTime,
-                                                     event.endTime,
-                                                     {
-                                                      location : event.location, 
-                                                      description : event.description
-                                                     });
+        var resultEvent;
+        if (event.isAllDay){
+          resultEvent = targetCalendar.createAllDayEvent(event.title,
```
Comment (michaelcadilhac) [inline Code.gs]: Where is `createAllDayEvent` implemented?

File: Code.gs [hunk 3]
```diff
@@ -161,6 +174,8 @@ function main(){
           
         if (emailWhenAdded)
           GmailApp.sendEmail(email, "New Event Added", "New event added to calendar \"" + targetCalendarName + "\" at " + event.startTime);
+          
+        Utilities.sleep(250); //Solve some "Creating too many events in a short time" issues
```
Comment (michaelcadilhac) [inline Code.gs]: This seems unrelated to handling all day events, isn't it?

File: Code.gs [hunk 4]
```diff
@@ -230,6 +245,9 @@ function ConvertToCustomEvent(vevent){
   var dtstart = vevent.getFirstPropertyValue('dtstart');
   var dtend = vevent.getFirstPropertyValue('dtend');
   
+  if (dtstart.isDate && dtend.isDate)
```
Comment (michaelcadilhac) [inline Code.gs]: How does that work?

File: Code.gs (hunk 1 above)
Comment (derekantrican) [reply Code.gs]: Possibly, but I have a feeling we would get more complaints of "it's not creating any events" [lots of troubleshooting later...] "oh, the calendar just wasn't shown in Google Calendar"

File: Code.gs (hunk 2 above)
Comment (derekantrican) [reply Code.gs]: Here: https://developers.google.com/apps-script/reference/calendar/calendar#createAllDayEvent(String,Date,Date,Object)

File: Code.gs (hunk 3 above)
Comment (derekantrican) [reply Code.gs]: Sure. Do you think this should be separate, then?

File: Code.gs (hunk 4 above)
Comment (derekantrican) [reply Code.gs]: Comes from here: http://mozilla-comm.github.io/ical.js/api/ICAL.Time.html (description: `If true, the instance represents a date (as opposed to a date-time)`). Basically, the `DTSTART` (or `DTEND`) object on the `VEVENT` can either be a date or datetime. You can see the discussion about that here: https://github.com/mozilla-comm/ical.js/issues/353

**Comment by derekantrican** [comment]
@michaelcadilhac Did you have any input on my replies?

**Comment by michaelcadilhac** [comment]
@derekantrican Aye, sorry for the lag, busy week.  Can I get back to you during the weekend?

**Comment by derekantrican** [comment]
@michaelcadilhac Sure, no prob

File: Code.gs (hunk 3 above)
Comment (michaelcadilhac) [reply Code.gs]: Well yes probably :-)

File: Code.gs (hunk 4 above)
Comment (michaelcadilhac) [reply Code.gs]: Ah fantastic, thanks for digging that up.

File: Code.gs (hunk 3 above)
Comment (derekantrican) [reply Code.gs]: Ok, this should be fixed now

**Comment by michaelcadilhac** [comment]
Note that this fails to modify events that are turned into all day events.

**Comment by derekantrican** [comment]
@michaelcadilhac Thanks for letting me know. Please add a new bug with any helpful information (eg see [this issue](https://github.com/derekantrican/GAS-ICS-Sync/issues/34) for good helpful information

**Comment by boillodmanuel** [comment]
Hello,

I added support for modifying all day events on my branch. As our codes has diverged, I can't open a pull request, but feel free to look at my code and reuse what you want : https://github.com/boillodmanuel/GAS-ICS-Sync/commit/4f72d13f168493bb40704fe93efbc354226bf29a

As far as I know, the main difference between our codes is the way we handle recurring events.

**Comment by derekantrican** [comment]
@boillodmanuel Good to know! I'll look at incorporating those changes. Currently I'm quite busy with other projects but hoping to get back into this one soon
